### PR TITLE
Rewrite search filter sidebar with React

### PR DESCRIPTION
**Pull request by cmfcmf**
I rewrote the search filters with React to more easily incorporate the following improvements:
- An "apply" button appears after changing a filter.
- Pressing ENTER inside an integer min/max field applies the filter
- Long filter lists now scroll when extended
- Other minor styling updates

The rewrite should also make it easier for a future PR to no longer load the huge list of options when the page is initially rendered and instead only fetch the big lists (e.g., list of notable items) in batches via AJAX.

This PR also showcases how modern JS like React can be integrated into AdventureLookup without much hassle.

### CODE REVIEW

**Comment by Wigginns** [comment]
Love this. React is cool stuff and this is a good step forward for the app imo.

What are your thoughts on approval processes? I usually tend to check out a branch and at least run it locally for a sanity check first but with the testing environment setup and CI/CD working again maybe that's overkill.

**Comment by cmfcmf** [comment]
Generally speaking I prefer to checkout PRs and test them locally, just to make sure they really work (especially because the tests don't cover that much). But I also know that it is always a hassle to do that. Maybe I'll find the time to setup Heroku Review Apps for AdventureLookup. That would make it easier to sanity check PRs, because you'd simply have to click a link to see a deployed version of the PR.

File: app/Resources/views/adventures/index.html.twig [hunk 1]
```diff
@@ -5,16 +5,8 @@
...
-                    <div class="content">
-                        <div class="sidebar-title">Adventure Lookup</div>
-                        <form method="post" action="{{ url('adventure_index') }}" id="search-form">
-                            <input type="hidden" value="{{ q }}" name="q" id="search-query-form" />
-                            {% include 'adventures/search_filters.html.twig' %}
-                        </form>
-                        <div id="filter-more"></div>
-                    </div>
-
+                <div class="adl-sidebar mt-5 mt-md-0" id="sidebar-react-root" data-fields="{{ fields.toArray | json_encode | e("html_attr") }}" data-url="{{ url('adventure_index') | e("html_attr") }}" data-initial-filter-values="{{ searchFilter | json_encode | e("html_attr")}}" data-initial-query="{{ q | e("html_attr") }}" data-field-stats="{{ stats | json_encode | e("html_attr") }}"></div>
```
Comment (Wigginns) [inline app/Resources/views/adventures/index.html.twig:8]: This really runs off the page. I don't know if there is a style guide I missed but maybe it makes sense to put each attribute on a new line? I'm showing my ignorance here 😅

Comment (cmfcmf) [reply app/Resources/views/adventures/index.html.twig]: Very true - fixed in c3eb30d. There isn't really a style guide in place right now.

**Pull request merged**